Assets/Scripts/GameManager.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class GameManager : MonoBehaviour
+{
+
+    public GameObject Player;
+    public List<GameObject> CheckPoints;
```

Comment:
`GameManager` does not need to know what the checkpoints are. Instead `Checkpoint`s inform `GameManager` when Player reaches the checkpoint.